pkg/fanal/secret/builtin-rules.go

```diff
@@ -783,4 +784,13 @@ var builtinRules = []Rule{
 		SecretGroupName: "secret",
 		Keywords:        []string{"typeform"},
 	},
+	{
+		ID:              "azure-data-factory-registration-key",
+		Category:        CategoryAzure,
+		Title:           "Azure Data Factory Self-Hosted Integration Runtime registration key",
+		Severity:        "HIGH",
+		Regex:           MustCompile(`IR@[0-9a-zA-Z-]{36}@(.*?)@`),
+		SecretGroupName: "secret",
+		Keywords:        []string{"azure_data_factory"},
```

Comment:
Regex uses a lot of resources. To save them we use next logic:
We try to find `Keyword` in file.
If `Keyword` is found => use `Regex` to find secret.
else => skip file.

Thus `Regex` should include `Keyword`.
Something like this:
```suggestion
		Keywords:        []string{"IR@"},
```